python/tests/test_util.py

```diff
@@ -540,3 +540,27 @@ def test_length_bad_value(self, length):
     def test_length_bad_type(self, length):
         with pytest.raises(TypeError, match="argument must be a string"):
             tskit.random_nucleotides(length)
+
+
+class TestAlleleRemap:
+    a = ["A", "C", "G", "T"]
+    b = ["G", "C"]
+    c = []
```

Comment:
It would be good to add some more tests here you could use `@pytest.mark.parametrize` as in `test_reference_sequence:179`. Some ideas:
 - What happens if one/both the arguments are a single string?
 - Error states such as an argument being `None`, these tests help us understand what exceptions are thrown by a method and that they don't silently do something unwanted.

Reply:
By "one/both the arguments are a single string", do you mean if at least one of the inputs is not a list, just a string?

Reply:
Thanks, I get what you mean now!

Reply:
Maybe we can check that both the input lists are of the same type and are one of list, tuple or string.

Reply:
I don't think we need to be overly restrictive - they don't need to be the same type. Also I almost forgot - can we add some numpy examples to the tests?

Reply:
The function is using index(), which numpy arrays don't have. Maybe convert a numpy array into a Python list?

Reply:
Ah! You mentioned below!

Reply:
> The function is using `index()`, which numpy arrays don't have. Maybe convert a numpy array into a Python list?

Probably better to do it the other way around and convert to numpy array at the start, then we support all sequences that numpy does.

Reply:
So, the inputs can be a Numpy array or any Python Sequence-type object. And the output is a Numpy array (or a Python list)?

Reply:
Also, we are allowing for duplicate elements in either or both of the inputs?

Reply:
I've updated the function. Now, the inputs can be a numpy array or any Python Sequence-type object, and the output is a numpy array.

Reply:
When the input is a string, then it is treated as a list with one element, which is the string itself.